lib/matplotlib/backends/_backend_tk.py

```diff
@@ -96,7 +96,16 @@ def _on_timer(self):
         # if _timer is None, this means that _timer_stop has been called; so
         # don't recreate the timer in that case.
         if not self._single and self._timer:
-            self._timer = self.parent.after(self._interval, self._on_timer)
+            if self._interval:
```

Comment:
```suggestion
            if self._interval > 0:
```

Because conceptually we have an int comparison and not a Boolean check.